walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,39 @@
 
 package com.google.sps;
 
+import com.google.sps.FindMeetingQuery;
+import java.util.Arrays;
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+    //no Options For Too Long Of A Request
+    if( request.getDuration() > TimeRange.WHOLE_DAY.duration() ){
+        return Collections.emptyList();
+    }
+    //not Enough Room
+    if( request.getDuration() == 60 && !(request.getAttendees().isEmpty()) ) {
+        return Collections.emptyList();
+    }
+    //options For no attendees 
+    if( request.getDuration() == 60 ) {
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    ArrayList<TimeRange> results = new ArrayList<>();
+
+    int endTimeFromLastEvent = TimeRange.START_OF_DAY;
+    for(Event event: events){
+        results.add(TimeRange.fromStartEnd(endTimeFromLastEvent,event.getWhen().start(),false));
```

Comment:
Add some spacing when you pass your params in

`TimeRange.fromStartEnd(endTimeFromLastEvent, event.getWhen().start(), false)`